source/renderer/app/components/wallet/settings/WalletSettings.js

```diff
@@ -172,6 +217,81 @@ export default class WalletSettings extends Component<Props, State> {
     );
   };
 
+  onUndelegateWalletClick = async () => {
+    const {
+      walletId,
+      openDialogAction,
+      updateDataForActiveDialogAction,
+    } = this.props;
+    this.onBlockForm();
+    openDialogAction({
+      dialog: UndelegateWalletConfirmationDialog,
+    });
+    updateDataForActiveDialogAction({
+      data: { walletId },
+    });
+  };
+
+  renderUndelegateWalletBox = () => {
+    const { intl } = this.context;
+    const {
+      delegationStakePoolStatus,
+      isRestoring,
+      isLegacy,
+      walletReward,
+      isDialogOpen,
+      onDelegateClick,
+      undelegateWalletDialogContainer,
+    } = this.props;
+    const notDelegating =
+      delegationStakePoolStatus === WalletDelegationStatuses.NOT_DELEGATING;
+
+    if (!IS_WALLET_UNDELEGATION_ENABLED) {
+      return null;
+    }
+
+    if (isLegacy || (!notDelegating && !walletReward.isZero())) {
+      return null;
+    }
```

Comment:
This can be one `if` clause

Reply:
@tomothespian I did this intentionally because we will remove config variable eventually.
Wanted to avoid editing valid if clause in 253 line when removing config.

Reply:
@yakovkaravelov aha I see. Then we are good with this one. Resolved!